### PR TITLE
Use Index Scan for EXACT and PREFIX pattern matches for LIKE for Const node inside CollateExpr

### PR DESCRIPTION
**Pull request by ahmed-shameem**
### Description

With this commit: https://github.com/babelfish-for-postgresql/babelfish_extensions/commit/e119968d7ffe59a49f2326d5fc421b1442d9c40c, babelfish now use Index Scans for the following cases:

-- CASE 1
```
SELECT COL FROM TAB WHERE COL LIKE 'ab'
```

-- CASE 2
```
SELECT COL FROM TAB WHERE COL LIKE 'a%'
```

However if there is a Collate clause with the right operand :
-- CASE 1
```
SELECT COL FROM TAB WHERE COL LIKE 'ab' COLLATE DATABASE_DEFAULT
```

-- CASE 2
```
SELECT COL FROM TAB WHERE COL LIKE 'a%' COLLATE DATABASE_DEFAULT
```
babelfish does NOT use Index Scan.

This commit enables Index scan for such cases. We do this by removing CollateExpr on both operands as those are redundant and the OpExpr (LIKE) operator is already aware of the collation it needs to use. 

### Issues Resolved

BABEL-5077
Signed-off-by: Shameem Ahmed [shmeeh@amazon.com](mailto:shmeeh@amazon.com)

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by forestkeeper — DISMISSED** [review]
Lgtm

File: contrib/babelfishpg_tsql/src/collation.c [hunk 1]
```diff
@@ -386,6 +386,20 @@ optimise_likenode(Node *node, OpExpr *op, like_ilike_info_t like_entry, coll_inf
 		op->opfuncid = like_entry.ilike_opfuncid;
 	}
 	
+	op->inputcollid = tsql_get_oid_from_collidx(collidx_of_cs_as);
+
+
+	/* Remove CollateExpr as the op->inputcollid has already been set */
+	if (IsA(rightop, CollateExpr))
+	{
+		lsecond(op->args) = rightop = (Node*)((CollateExpr*) rightop)->arg;
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/src/collation.c:395]: we need to copy rightarg on 342

File: contrib/babelfishpg_tsql/src/collation.c [hunk 2]
```diff
@@ -870,7 +883,17 @@ convert_node_to_funcexpr_for_like(Node *node, Oid inputcollid)
...
+					#ifdef USE_ICU
+						if (U_ICU_VERSION_MAJOR_NUM == pltsql_remove_accent_map_icu_major_version && U_ICU_VERSION_MINOR_NUM == pltsql_remove_accent_map_icu_min_version)
+						{
+							con->constvalue = DirectFunctionCall1(remove_accents_internal_using_cache, con->constvalue);
+							con->constcollid = InvalidOid;
+							return (Node *) con;
+						}
+					#endif
 					con->constvalue = DirectFunctionCall1(remove_accents_internal, con->constvalue);
+					con->constcollid = InvalidOid;
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/src/collation.c]: we can use `OidFunctionCall1` with remove_accents_internal_oid

**Review by Deepesh125 — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/src/collation.c (hunk 1 above)
Comment (ahmed-shameem) [reply contrib/babelfishpg_tsql/src/collation.c:395]: Updated in latest rev

File: contrib/babelfishpg_tsql/src/collation.c (hunk 2 above)
Comment (ahmed-shameem) [reply contrib/babelfishpg_tsql/src/collation.c]: Updated in latest rev

**Review by Deepesh125 — APPROVED** [review]
